packages/server-core/src/setting/feature-flag-setting/migrations/20241105001200_primary_flagNames.ts

```diff
@@ -0,0 +1,144 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import {
+  FeatureFlagSettingType,
+  featureFlagSettingPath
+} from '@ir-engine/common/src/schemas/setting/feature-flag-setting.schema'
+import type { Knex } from 'knex'
+import { v4 as uuidv4 } from 'uuid'
+
+type OldFeatureFlagSettingType = FeatureFlagSettingType & {
+  flagName: string
+}
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export async function up(knex: Knex): Promise<void> {
+  // as in "upgrade"
+
+  await knex.raw('SET FOREIGN_KEY_CHECKS=0')
+
+  migrateUp: {
+    const flagNameColumnExists = await knex.schema.hasColumn(featureFlagSettingPath, 'flagName')
+
+    if (!flagNameColumnExists) {
+      break migrateUp
+    }
+
+    const oldRows = await knex.select<OldFeatureFlagSettingType[]>().from(featureFlagSettingPath)
```

Comment:
rather than persisting the data in memory and recreating the table, we can just migrate the columns in question. @hanzlamateen do you have any opinions on this?

Reply:
Yes @Rezmason, I think we can use alter table. And then makes the updates. renameColumn and dropColumn.

Reference:
https://github.com/theinfinitereality/irpro-multitenancy/blob/97dfa0194905d9e87e3a354d17517a274632b243/services/account/account-slug-generator/migrations/20240919180000_rename_customerId.ts#L14

https://github.com/ir-engine/ir-engine/blob/b1995ec4135d23901f4f3cda15dddb55d67fa6ee/packages/server-core/src/matchmaking/match-instance/migrations/20230425043432_match-instance.ts#L43